mutations/loadLock.js

```diff
@@ -70,117 +113,189 @@ async function loadLock(inputs, models, req) {
-                validationErrors.push("The keyholder does not allow test locks");
-           }
-       }
+        if(NumOfLocksLoaded.length >= UserLimit) {
+            validationErrors.push("The keyholder has set a limit on how many of these locks that can be loaded and the limit has been reached.");
+        }
+    }
+    // tested including boundary
 
-       if(LockSearch.Block_User_Rating_Enabled) {
-           const MinRating = LockSearch.Block_User_Rating
+    if(LockSearch.Block_Test_Locks) {
+        if(inputs.Test_Lock === true) {
+            validationErrors.push("The keyholder does not allow test locks");
+        }
+    }
+    // tested
 
-           if(await getLockeeRating(req.Authenticated) < MinRating) {
-                validationErrors.push("The keyholder needs you to have a higher lockee rating before loading this lock");
-           }
-       }
+    if(LockSearch.Block_User_Rating_Enabled) {
+        const MinRating = LockSearch.Block_User_Rating
+        if (MinRating === undefined) { // should never happen, but...
+            throw new Error("CreatedLock object blocks user based on rating, but doesn't specify min rating")
+        }
+        if(await getLockeeRating(req.Authenticated) < MinRating) {
+            validationErrors.push("The keyholder needs you to have a higher lockee rating before loading this lock");
+        }
+    }
+    // tested including boundary
 
-       if(LockSearch.Block_Already_Locked) {
-           const AlreadyLocked = await models.LoadedLock.findOne({
-               where: {
-                   Lockee: req.Authenticated,
-                   Unlocked: false
-               }
-           });
-           if(AlreadyLocked != null) {
-                validationErrors.push("The keyholder does not allow you to load this alongside other locks");
-           }
-       }
-       
-        if(LockSearch.Block_Stats_Hidden) {
-            const UserSettings = await models.UserSetting.findOne({
-                where: {
-                    User_ID: req.Authenticated
-                }
-            });
-            if(UserSettings.Share_Stats === false) {
-                validationErrors.push("The keyholder requires that you share your stats");
+    if(LockSearch.Block_Already_Locked) {
+        const AlreadyLocked = await models.LoadedLock.findOne({
+            where: {
+                Lockee: req.Authenticated,
+                Unlocked: false
             }
+        });
+        if(AlreadyLocked != null) {
+            validationErrors.push("The keyholder does not allow you to load this alongside other locks");
         }
+    }
+    // tested
 
-        if(LockSearch.Require_DM === true) {
-            if(inputs.Sent_DM != true) {
-                validationErrors.push("The keyholder requires that you speak to them prior to loading this lock. Please refer to where you found the lock for more details.");
+    // TODO: ??? possibly ??? CK also prevents any lock from loading if 
```

Comment:
yes this should be the case. But we probably also need to think about exceptions:
- allowing different locks of the same keyholder to be loaded (e.g. for different things / bondage)
- maybe allowing bondage or task locks to be loaded in the future? Is just a heads up, nothing for now